src/daemon/ca/manager.rs

```diff
@@ -231,158 +255,178 @@ impl CaServer {
-    /// Note: typically a CA will use only one repository, but during migrations there may be multiple.
-    pub async fn ca_repo_elements(&self, ca: &Handle) -> KrillResult<HashMap<RepositoryContact, Vec<PublishElement>>> {
-        Ok(self.ca_objects_store.ca_objects(ca)?.elements())
-    }
-
-    /// Get all old repos for best effort clean-up. They may not be reachable after all.
-    /// Clears the list of old repos for the given CA.
-    pub fn ca_take_deprecated_repos(&self, ca: &Handle) -> KrillResult<Vec<RepositoryContact>> {
-        let mut res = vec![];
-
-        self.ca_objects_store.with_ca_objects(ca, |objects| {
-            res = objects.ca_take_deprecated_repos();
+/// # CA instances and identity
+///
+impl CaManager {
+    /// Initializes a CA without a repo, no parents, no children, no nothing
+    pub fn init_ca(&self, handle: &Handle) -> KrillResult<()> {
+        if handle == &ta_handle() || handle.as_str() == "version" {
+            Err(Error::TaNameReserved)
+        } else if self.ca_store.has(handle)? {
+            Err(Error::CaDuplicate(handle.clone()))
+        } else {
+            // Initialize the CA in self.ca_store, but note that there is no need to create
+            // a new CA entry in self.ca_objects_store or self.status_store, because they will
+            // generate empty default entries if needed.
+            let init = IniDet::init(handle, self.signer.deref())?;
+            self.ca_store.add(init)?;
             Ok(())
-        })?;
-
-        Ok(res)
+        }
     }
 
-    /// Update repository where a CA publishes.
-    pub async fn update_repo(&self, handle: Handle, new_contact: RepositoryContact, actor: &Actor) -> KrillResult<()> {
-        let cmd = CmdDet::update_repo(&handle, new_contact, self.signer.clone(), actor);
+    /// Updates the self-signed ID certificate for a CA. Use this with care as
+    /// RFC 8183 only talks about initial ID exchanges in the form of XML files.
+    /// It does not talk about updating identity certificates and keys. Krill supports
+    /// that a new ID key pair and certificate is generated, and has functions to update
+    /// this for a parent, a child, a repo and a publisher, but other implementations may
+    /// not support that identities are updated after initialization.
+    pub async fn ca_update_id(&self, handle: Handle, actor: &Actor) -> KrillResult<()> {
+        let cmd = CmdDet::update_id(&handle, self.signer.clone(), actor);
         self.send_command(cmd).await?;
         Ok(())
     }
 
-    /// Returns the RepoStatus for a CA
-    pub async fn ca_repo_status(&self, ca: &Handle) -> KrillResult<RepoStatus> {
-        let lock = self.locks.ca(ca).await;
-        let _ = lock.read().await;
-        if self.ca_store.has(ca)? {
-            self.status_store.lock().await.get_repo_status(ca).await
-        } else {
-            Err(Error::CaUnknown(ca.clone()))
-        }
+    /// Get the current CAs
```

Comment:
I'm not sure what is meant by "current" here. Perhaps also note that the list of CAs returned will not contain CAs that the given actor is not permitted to know about.